champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/sword/Riposte.java

```diff
@@ -77,118 +83,142 @@ public SkillType getType() {
-                riposting.put(target.getName(), (long) (System.currentTimeMillis() + ((level * 0.5)) * 1000));
-                event.cancel("Riposte");
+        Particle.SMOKE_LARGE.builder().location(player.getLocation().add(0, 0.25, 0)).receivers(20).extra(0).spawn();
+    }
 
+    @UpdateEvent(delay = 100)
+    public void onUpdateEffect() {
+        Iterator<UUID> it = active.iterator();
+        while (it.hasNext()) {
+            Player player = Bukkit.getPlayer(it.next());
+            if (player != null) {
+                if (player.isHandRaised()) {
+                    player.getWorld().playEffect(player.getLocation(), Effect.STEP_SOUND, Material.IRON_BLOCK);
+                }
             } else {
+                it.remove();
+            }
+        }
+    }
 
-                if (riposting.containsKey(target.getName())) {
-
-                    long time = riposting.get(target.getName()) - System.currentTimeMillis();
-                    double remaining = UtilTime.convert(time,
-                            UtilTime.TimeUnit.SECONDS, 1);
-
-                    if (time <= 0) {
-                        riposting.remove(target.getName());
-                        return;
+    @UpdateEvent
+    public void onUpdate() {
+        Iterator<UUID> it = active.iterator();
+        long currentTime = System.currentTimeMillis();
+
+        while (it.hasNext()) {
+            Player player = Bukkit.getPlayer(it.next());
+            if (player != null) {
+                int level = getLevel(player);
+                if (level > 0) {
+
+                    if (player.isHandRaised() && !handRaisedTime.containsKey(player)) {
+                        handRaisedTime.put(player, System.currentTimeMillis());
                     }
 
-                    if (damager instanceof Player player) {
-                        UtilMessage.simpleMessage(player, getClassType().getName(),
-                                "<yellow>%s<gray> is resistant to melee attacks for <green>%.1f<gray> seconds.", target.getName(), remaining);
+                    if (!player.isHandRaised() && handRaisedTime.containsKey(player) && !boostedDamage.containsKey(player)) {
+                        handRaisedTime.remove(player);
+                        it.remove();
+                        UtilMessage.message(player, getClassType().getName(), "Your Riposte failed.");
+                        player.getWorld().playSound(player.getLocation(), Sound.UI_BUTTON_CLICK, 2.0f, 1.0f);
                     }
 
-                    target.getWorld().playSound(target.getLocation(), Sound.ENTITY_ZOMBIE_ATTACK_IRON_DOOR, 1.0F, 1.0F);
-                    event.setDamage(event.getDamage() * (1 - damageReduction));
-                    event.setKnockback(false);
+                    if (player.isHandRaised() && handRaisedTime.containsKey(player) && UtilTime.elapsed(handRaisedTime.get(player), (duration * 1000)) && !boostedDamage.containsKey(player)) {
```

Comment:
```suggestion
                    if (player.isHandRaised() && handRaisedTime.containsKey(player) && UtilTime.elapsed(handRaisedTime.get(player), (long) (duration * 1000)) && !boostedDamage.containsKey(player)) {
```

Reply:
I love the part where he makes a PR for something that doesn't even compile

Reply:
really mad